src/bsd.rs

```diff
@@ -150,18 +166,19 @@ fn if_name_mtu(idx: u32) -> Result<(String, usize)> {
             .to_str()
             .map_err(|err| Error::new(ErrorKind::Other, err))?
     };
-
+    let mut mtu = None;
     for ifa in IfAddrs::new()?.iter() {
         if ifa.addr().sa_family == AF_LINK && ifa.name() == name {
             if let Some(ifa_data) = ifa.data() {
-                if let Ok(mtu) = usize::try_from(ifa_data.ifi_mtu) {
-                    return Ok((name.to_string(), mtu));
+                if let Ok(ifi_mtu) = usize::try_from(ifa_data.ifi_mtu) {
+                    mtu = Some(ifi_mtu);
+                    break;
                 }
             }
-            return Err(default_err());
+            break;
         }
     }
-    Err(default_err())
+    Ok((name.to_string(), mtu))
```

Comment:
The nesting is difficult for me to parse. What do you think of the following? (Not tested.)

``` rust
    let mtu = IfAddrs::new()?
        .iter()
        .find(|ifa| ifa.addr().sa_family == AF_LINK && ifa.name() == name)
        .and_then(|ifa| ifa.data)
        .and_then(|if_data| usize::try_from(ifa_data.ifi_mtu).ok());
    Ok((name.to_string(), mtu))

```